tools/SourceKit/tools/sourcekitd/lib/Service/Requests.cpp

```diff
@@ -1060,9 +1261,17 @@ static void handleSemanticRequest(
     Optional<RequestDict> options = Req.getDictionary(KeyCodeCompleteOptions);
     return Rec(codeCompleteOpen(*Name, InputBuf.get(), Offset, options, Args,
                                 std::move(vfsOptions), CancellationToken));
-  }
+  });
+}
+
+static void
+handleRequestCodeCompleteUpdate(const RequestDict &Req,
+                                SourceKitCancellationToken CancellationToken,
+                                ResponseReceiver Rec) {
+  if (checkVFSNotSupported(Req, Rec))
+    return;
```

Comment:
Previously, rejecting VFS options is done by "allow" list at the top of `handleRequestImpl` and allowed only 4 request kinds, that was often overlooked. Even if the request allowed VFS options, the request failed.

Now it's an opt-out option handled in each request handling function.